### ISSUE TITLE
Increase size of known DS network

### ISSUE THREAD

**Issue by Chronial** [Wulf2k/DaS-PC-MPChan#106]
At the moment we know about 1700 of the 4000 nodes online. This could be better. I had this idea on how we can improve that number:

* Remember the steamid of all non-dscm nodes for a month
* Every 10 minutes, ask steam who of these nodes (minus the ones we see in DSCM-Net) is currently playing dark souls.
* Add these nodes to a WANT list.
* Reserve one slot on every client for "network-size-increase", hand out the nodes on the WANT list to DSCM clients.
* Once we have details on a node, other DSCMs should connect to it because it's a useful node to them, thus freeing up the WANT slot for more discovery.

Maybe we should also make DSCM more reluctant to disconnect from non-dscm nodes that only we are connected to.

**Comment by Wulf2k** [comment]
I like it.  Though if somebody's sitting at the menu, we wouldn't want them to get connected to every 5 minutes by people working their way through the WANT list.  Or maybe we do.

And we'd need to be sure somehow that clients were actually trying their WANT list.  If their nodes are full of people that won't get removed any time soon, there could be many entries that never actually get checked.

Nothing too complex to overcome and sounds like a useful feature.  It would even be another step towards fixing Dark Souls Multiplayer for people who aren't even using our mod.  If they're online and stuck at 0 nodes, they can still get pulled in by people working their way through the list.

A more enthusiastic person might even post something like "LINK THE FIRE!"

**Comment by Chronial** [comment]
> I like it. Though if somebody's sitting at the menu, we wouldn't want them to get connected to every 5 minutes by people working their way through the WANT list. Or maybe we do.

I hope that less then 5% of DS players spend the majority of their time in the loading screen, so I don't think we need to care about this.

> And we'd need to be sure somehow that clients were actually trying their WANT list. If their nodes are full of people that won't get removed any time soon, there could be many entries that never actually get checked.

We have a central server that can easily handle all of this. The client doesn't have a WANT list – the server does.

>  It would even be another step towards fixing Dark Souls Multiplayer for people who aren't even using our mod.

Yes – this would probably completely fix multiplayer for anyone who has seen any nodes in the last month.

**Comment by Wulf2k** [comment]
> The client doesn't have a WANT list – the server does.

Right, phrased poorly, but I mean that clients are trying their WANT slots.

Though I was just looking at it from the wrong angle, there's no reason a client should sit on a WANTed node if they have no slots free.

**Comment by Chronial** [comment]
This is all going to be somewhat fuzzy anyways. The nodes might just not be reachable because of a firewall etc. I would just have a list a hand out nodes in a round-robin fashion.
